tests/regression/roles/sonic_poe/defaults/main.yml

```diff
@@ -0,0 +1,125 @@
+---
+ansible_connection: httpapi
+module_name: sonic_poe
+
+tests:
+  - name: test_case_01_no_card_support
+    description: test merging all config
+    state: merged
+    input:
+      interfaces:
+        - name: "{{interface1}}"
+          detection: dot3bt
+          priority: low
+      global:
+        power_mgmt_model: 'class'
+
+  - name: test_case_02_all_card_settings
+    description: test merging all settings of additional cards
+    state: merged
+    input:
+      cards:
+        - card_id: 0
+          usage_threshold: 50
+          auto_reset: True
+          power_mgmt_model: dynamic
+
+  - name: test_case_03_all_interface_settings
+    description: test merging all settings for interfaces
+    state: merged
+    input:
+      interfaces:
+        - name: "{{interface2}}"
+          enabled: True
+          detection: legacy
+          disconnect_type: dc
+          four_pair: True
+          high_power: True
+          power_classification: normal
+          power_limit: 3000
+          power_limit_type: user-defined
+          power_pairs: spare
+          power_up_mode: dot3at
+          priority: low
+          use_spare_pair: True
+
+  - name: test_case_04_all_global_settings
+    description: test merging all global settings for poe
+    state: merged
+    input:
+      global:
+        auto_reset: True
+        power_mgmt_model: 'class'
+        usage_threshold: 49
+  
+  - name: test_case_05_delete_card
+    description: test deleting one item from cards list
+    state: deleted
+    input:
+      cards:
+        - card_id: 0
+
+  - name: test_case_06_delete_interface
+    description: test deleting one item from interfaces list
+    state: deleted
+    input:
+      interfaces:
+        - name: "{{interface2}}"
+
+  - name: test_case_07_delete_global
+    description: test deleting all settings from global poe settings
+    state: deleted
+    input:
+      global:
+        auto_reset: True
+        power_mgmt_model: 'class'
+        usage_threshold: 90
+  
```

Comment:
It would be good to add a test case that deletes all global configuration by specifying the 'global' keyword with no sub-attributes (if that is supported, which it apparently is).

Reply:
Although there is now a test case (test_case_03_delete_global) that explicitly deletes the only global configuration setting, the code also contains support for a "delete all" of global config:

```
        if "global" in want and "global" in have:
            if len(want["global"]) == 0:
                # allow specifying blank for global section causes delete all global
                deleted_global_settings = have["global"]
```

This is not being tested. Please verify that this handling is possible (not prevented by having the Ansible infrastructure filling in 'empty' configuration sub-dicts with keys having no associated value when a playbook specifies one sub-dict with values). If it is possible, then after test_case_03_delete_global, please add a test case that adds back the global config attribute, then another to verify that a 'deleted' playbook that specifies the 'global' key with no value also causes deletion of the global attribute.

If it is possible to support the 'delete all' capability for a sub-dict within this argspec, then the same capability should be added for the other sub-dicts ('interfaces', and 'cards') along with the corresponding test case for 'delete all' on the 'interfaces' sub-dict. (I realize that testing of this functionality is currently not possible for the 'cards' sub-dict.) If 'delete all' functionality isn't possible for any of the sub-dicts (for the reason stated in the previous paragraph), then the corresponding case coverage for the 'global' sub-dict should be removed.